### ISSUE TITLE
pass props to imported a component in tdComp

### ISSUE THREAD

**Issue by mubashar856** [OneWayTech/vue2-datatable#97]
I have assigned "tdOpt" component to tdComp now I am importing another component into "tdOpt" and also passing prop to that component, but it returns undefined.

table-component.vue

columns: [
                { title: 'Lower', field: 'lowerLimit', sortable: true},
                { title: 'Upper', field: 'upperLimit', sortable: true },
                { title: 'Value', field: 'value', sortable: true },
                { title: 'Calc. Type', field: 'calculationType' },
                { title: 'Operation', tdComp: 'tdOpt' }
            ] 

tdOpt.vue

<slab-form
                        @form-data-updated="setupSlabDetail"
                        @errors-updated="setErrors"
                        @form-resetted="formReset = false"
                        :slab="row" />

slabForm.vue

props: ['slab']

now this slab always return undefined.

**Comment by kenberkeley** [comment]
`props: ['row']`

**Comment by mubashar856** [comment]
but I am passing slab as prop for rowm check tdOpt.vue file code

**Comment by kenberkeley** [comment]
claim `props: ['row']` in tdOpt.vue

**Comment by kenberkeley** [comment]
https://onewaytech.github.io/vue2-datatable/doc/#/en/details/dynamic-comps?id=%E2%8A%99-tdcomp

**Comment by mubashar856** [comment]
yes and then I am passing that row prop to slab as prop in tdOpt.vue

**Comment by kenberkeley** [comment]
paste the full code here plz

**Comment by mubashar856** [comment]
```
<template>
        <div>
            <slab-form
            @form-data-updated="setupSlabDetail"
            @errors-updated="setErrors"
            @form-resetted="formReset = false"
            :slab="row" />
        </div>
    </template>
    <script>
    import SlabForm from './slabForm'
    export default {
        props: ['row'],
        components:{SlabForm}
    }
    </script>

```

now code of slabForm.vue is below

```
    <template>
        <div></div>
    </template>
    <script>
    export default {
        props: ['slab']
    }
    </script>
```

now this slab prop always show undefined in vue devtool

**Comment by mubashar856** [comment]
no I dont think so that I have reversed it.

my table is in table-component.vue as given below

```
columns: [
{ title: 'Lower', field: 'lowerLimit', sortable: true},
{ title: 'Upper', field: 'upperLimit', sortable: true },
{ title: 'Value', field: 'value', sortable: true },
{ title: 'Calc. Type', field: 'calculationType' },
{ title: 'Operation', tdComp: '**tdOpt**' }
]
```

now I am passing my tdOpt.vue component in tdComp as written above

code in tdOpt.vue is written below

 ```
<template>
        <div>
            <slab-form
            @form-data-updated="setupSlabDetail"
            @errors-updated="setErrors"
            @form-resetted="formReset = false"
            :slab="row" />
        </div>
    </template>
    <script>
    import SlabForm from './slabForm'
    export default {
        props: ['row'],
        components:{SlabForm}
    }
    </script>

```

now I am passing that row to a third component name slabForm.vue and code of slabForm.vue is below

```
    <template>
        <div></div>
    </template>
    <script>
    export default {
        props: ['slab']
    }
    </script>
```

now this slab prop always show undefined in vue devtool

**Comment by kenberkeley** [comment]
how about console `this.row` in tdOpt.vue?

**Comment by mubashar856** [comment]
it returns expected output

**Comment by kenberkeley** [comment]
so what you mean is: you passed a prop to a component but it can't receive it... thus it's not my problem...

**Comment by mubashar856** [comment]
yes exactly, I think this issue occurs when we pass a component in tdComp then in that component we import the third component and pass prop to that third component, then it shows undefined. 
I think you got my point

**Comment by kenberkeley** [comment]
can you see the `props` declaration in SlabForm.vue?

**Comment by mubashar856** [comment]
yes, when I console slab it returns undefined in SlabForm.vue

**Comment by kenberkeley** [comment]
sucks... no idea at all...

**Comment by mubashar856** [comment]
Still, find nothing even watch is not working on slab prop but when I assign that slab prop to another variable it shows value there. So it worked for me,
Thanks
